### DISCUSSION TITLE
Form.setValues 的 merge 行为不正确

### DISCUSSION THREAD

**Discussion by liuweiGL** [alibaba/formily#1443] [Q&A 2.x] [answered]
## CodeSandBox
https://codesandbox.io/s/little-framework-1x6y5?file=/App.tsx

## 问题

当前 merge 使用的是 deep merge 策略导致无法重置 object field 的值。

## 期望

只在 form values 这一层进行 merge，具体到某个字段的时候应该直接替换。

**Comment by janryWang** [comment, marked as answer]
这行为是正确的，你改成overwrite就行了

**Comment by liuweiGL** [comment]
Override 会把整个 values 替换掉，这也不是我想要的行为。

**Comment by liuweiGL** [comment]
```ts
import { createForm } from '../shared'

const data = {
  a: {
    b: 1,
  },
  c: 2,
}

const form = createForm({
  values: data,
})

form.setValues({
  a: {},
})

console.log(form.values)

// 期待输出：{ a: {}, c: 2 }

```

**Reply by janryWang** [reply]
两种解法，一种是合并策略支持浅合并，还有一种是自己手动浅合并
form.values = {...form.values,a:{}}

**Reply by liuweiGL** [reply]
我觉得这样不太好看啊。。。我先这么处理吧

**Comment by liuweiGL** [comment]
现在有个问题是 form 中有多个 object value，我没法使用 form.setValues 来批量重置值。

**Comment by liuweiGL** [comment]
使用深合并是基于什么原因呢，感觉日常开发中浅合并使用的频率更高吧？
